### PR TITLE
Feature/right req

### LINKED ISSUE #95: As an organization admin, I want to get borrow requests to my organization only, so that I see relevant requests for me
- [ ] Hide all borrow requests to groups the user isn't part of

### PR DESCRIPTION
**Pull request by EmilMadsstuen**
Users can now only see burrow requests for games that they own or for games administrated by organizations where they are admins. Fixes #95

### CODE REVIEW

**Review by Supergamer1337 — CHANGES_REQUESTED** [review]

File: frontend/src/components/BorrowRequestsList/BorrowRequestsList.tsx [hunk 1]
```diff
@@ -15,7 +15,6 @@ interface BorrowRequest {
 const BorrowRequestsList: FC<BorrowRequestsListProps> = () => {
 	const [apiPath, setApiPath] = useState("/borrow/request/list")
 	const { data, error, loading } = useApiGet<BorrowRequest[]>(apiPath);
-
 	return (
```
[the remark this answers is not in the document]
Comment (x183) [reply frontend/src/components/BorrowRequestsList/BorrowRequestsList.tsx:18]: No! :triumph:

File: frontend/src/components/BorrowRequestsList/BorrowRequestsList.tsx [hunk 2]
```diff
@@ -15,7 +15,6 @@ interface BorrowRequest {
 const BorrowRequestsList: FC<BorrowRequestsListProps> = () => {
 	const [apiPath, setApiPath] = useState("/borrow/request/list")
 	const { data, error, loading } = useApiGet<BorrowRequest[]>(apiPath);
```
Comment (Supergamer1337) [inline frontend/src/components/BorrowRequestsList/BorrowRequestsList.tsx:17]: Keep line or I angry
```suggestion
	const [apiPath, setApiPath] = useState("/borrow/request/list")
	const { data, error, loading } = useApiGet<BorrowRequest[]>(apiPath);
	
```

File: backend/src/routers/borrowRequestRouter.ts [hunk 3]
```diff
@@ -171,10 +172,14 @@ borrowRequestRouter.post(
...
-borrowRequestRouter.get('/list', async (_, res) => {
-	const requests = await getActiveBorrowRequests();
-	res.status(200).json(await formatBorrowRequests(requests));
-});
+borrowRequestRouter.get(
+	'/list',
+	async (req, res) => {
+		const requests = await getActiveBorrowRequests(await getAccountFromCid((req.user as GammaUser).cid));
+		res.status(200).json(await formatBorrowRequests(requests));
+	}
```
[the remark this answers is not in the document]
Comment (EmilMadsstuen) [reply backend/src/routers/borrowRequestRouter.ts:180]: Should I catch the error maybe? I think it should be impossible to see the requests if you are not logged in.

File: backend/src/services/borrowRequestService.ts [hunk 4]
```diff
@@ -88,22 +88,59 @@ export const respondBorrowRequest = async (
 	return approved ? BorrowState.Approved : BorrowState.Rejected;
 };
 
-// TODO: Only get requests for game manager once implemented
-export const getActiveBorrowRequests = async () => {
-	return await prisma.borrow.findMany({
+
+export const getActiveBorrowRequests = async (account: any) => {
+	const organizationMemberships = await prisma.organizationMember.findMany({ //Get all the orgz where login is admin
```
[the remark this answers is not in the document]
Comment (EmilMadsstuen) [reply backend/src/services/borrowRequestService.ts]: Should be fixed now

File: backend/src/routers/borrowRequestRouter.ts (hunk 3 above)
[the remark this answers is not in the document]
Comment (EmilMadsstuen) [reply backend/src/routers/borrowRequestRouter.ts:180]: @Supergamer1337 I can't find a good solution. Do you have any tips?

**Review by EmilMadsstuen — COMMENTED** [review]
Did it a bit quickly. Does not work...

File: backend/src/routers/borrowRequestRouter.ts (hunk 3 above)
[the remark this answers is not in the document]
Comment (Supergamer1337) [reply backend/src/routers/borrowRequestRouter.ts:180]: There should be an `isAuthenricated` middleware that you can use. Just search for `isAuthenticated` in the code base for an example of how to use (I would suggest it, but I'm on my phone)

File: backend/src/routers/borrowRequestRouter.ts [hunk 5]
```diff
@@ -171,10 +172,14 @@ borrowRequestRouter.post(
  *
  * @apiUse ZodError
  */
-borrowRequestRouter.get('/list', async (_, res) => {
-	const requests = await getActiveBorrowRequests();
-	res.status(200).json(await formatBorrowRequests(requests));
-});
+borrowRequestRouter.get(
+	'/list',
```
Comment (x183) [inline backend/src/routers/borrowRequestRouter.ts:176]:
```suggestion
	'/list',
 isAuthenticated,
```

**Comment by x183** [comment]
Merge conflict! 😭

File: backend/src/routers/borrowRequestRouter.ts [hunk 6]
```diff
@@ -170,10 +171,17 @@ borrowRequestRouter.post(
...
+ isAuthenticated,
+	async (req, res) => {
+  const user = await getAccountFromCid((req.user as GammaUser).cid);
+  if(!user) return;
+		const requests = await getActiveBorrowRequests();
```
Comment (Oscariremma) [inline backend/src/routers/borrowRequestRouter.ts:180]: We need to pass the user
```suggestion
		const requests = await getActiveBorrowRequests(user);
```

File: backend/src/routers/borrowRequestRouter.ts [hunk 7]
```diff
@@ -141,9 +143,8 @@ borrowRequestRouter.post(
 				},
 				'Body'
 			);
-		let requestResponse = `Request ${
-			body.approved ? 'accepted' : 'rejected'
-		} successfully`;
+		let requestResponse = `Request ${body.approved ? 'accepted' : 'rejected'
```
Comment (Oscariremma) [inline backend/src/routers/borrowRequestRouter.ts:146]:
```suggestion
		const requestResponse = `Request ${body.approved ? 'accepted' : 'rejected'
```

**Review by Oscariremma — CHANGES_REQUESTED** [review]
We need to run prettier before merging as well

**Review by Oscariremma — APPROVED** [review]
LGTM

**Review by Supergamer1337 — APPROVED** [review]
LGTM

**Pull request merged**